core/src/main/java/org/apache/brooklyn/core/mgmt/ha/BrooklynBomOsgiArchiveInstaller.java

```diff
@@ -96,8 +96,21 @@ public class BrooklynBomOsgiArchiveInstaller {
     private boolean force = false;
     private boolean deferredStart = false;
     private boolean validateTypes = true;
-    
-    private File zipFile;
+
+    public static class FileWithTempInfo<T extends File> {
+        public FileWithTempInfo(T f, boolean isTemp) { this.f = f; this.isTemp = isTemp; }
+        public T f;
```

Comment:
I think `f` should be final to avoid it to be replaced. 
It's not very importand, but I'd feel more comfortable if instead of been public, it's exposed by a getter method, it also will make this code more clear:

```
FileWithTempInfo tempFile = new FileWithTempInfo(file, true);

...
tempFile.getFile();
_vs_
tempFile.f;
```